### PR TITLE
[DESK-704]filter and sort

### PR DESCRIPTION
**Pull request by mauricio0000**
### Description

Add feature sort and filter

### Checklist

- [ ] Pull Request title is in the format `[PROJ-12345] Some useful description here...`
- [ ] I have prefixed this PR with `[WIP]` if it is a Work In Progress (not ready to merge)
- [ ] The git branch is in the format `PROJ-12345-some-useful-description-here`
- [ ] I have added one or more reviewers
- [ ] I have added/updated tests for any code changes
- [ ] I have updated documentation (including README, inline comments and docs.remote.it docs)
- [ ] This PR only contains one isolated change (bug fix, feature, etc)
- [ ] Translation strings added/updated for all user-visible text
- [ ] I have manually reviewed this PR to make sure only the files I intended to change were changed and nothing else
- [ ] I have notified reviewers on Slack by @ mentioning them in the `#desktop`channel
- [ ] I have moved the Jira ticket into `Resolved` state and made a note in the ticket with a link to this PR

### Test plan

1. Sign in
2. Go to devices
3. You should see the new section of filters and sort

### Ticket

https://remoteit.atlassian.net/browse/DESK-756

### CODE REVIEW

**Comment by mauricio0000** [comment]
Miss sort by color and owner, waiting for changes API

File: frontend/src/components/DeviceList/DeviceList.tsx [hunk 1]
```diff
@@ -1,30 +1,40 @@
...
 
+  const css = useStyles()
+
   return (
-    <>
-      <List>
+    <div className={css.root}>
+      <List
+        className={clsx(css.content, {
+          [css.contentShift]: open,
```
Comment (JamieRuderman) [inline frontend/src/components/DeviceList/DeviceList.tsx:35]: We need to replace the <Drawer> MuiComponent with probably a <Collapse> one. Then if in a Flexbox it should all just naturally move over.

File: frontend/src/components/FilterDrawerContent/FilterDrawerContent.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      filter: state,
+      sort: sort,
+      owner: owner,
+      from: 0,
+    })
+    devices.fetch()
+  }
+
+  return (
+    <Drawer
```
Comment (JamieRuderman) [inline frontend/src/components/FilterDrawerContent/FilterDrawerContent.tsx:52]: Let's change this to a <Collapse> and see if we can remove all the extra code trying to make the Drawer work inside a container.

File: frontend/src/pages/DevicesPage/DevicesPage.tsx [hunk 3]
```diff
@@ -24,6 +25,13 @@ export const DevicesPage = () => {
       return lookup
     }, {}),
   }))
+
+  const [open, setOpen] = useState(false)
+
+  const handleDrawer = (state: boolean) => {
```
Comment (JamieRuderman) [inline frontend/src/pages/DevicesPage/DevicesPage.tsx:31]:
```suggestion
  const onOpen = (state: boolean) => {
```

File: frontend/src/pages/DevicesPage/DevicesPage.tsx [hunk 4]
```diff
@@ -49,7 +57,7 @@ export const DevicesPage = () => {
       ) : !devices.length ? (
         <DeviceListEmpty />
       ) : (
-        <DeviceList devices={devices} connections={connections} />
+        <DeviceList devices={devices} connections={connections} handleDrawer={handleDrawer} open={open} />
```
Comment (JamieRuderman) [inline frontend/src/pages/DevicesPage/DevicesPage.tsx:60]:
```suggestion
        <DeviceList devices={devices} connections={connections} onOpen={onOpen} open={open} />
```

File: frontend/src/pages/DevicesPage/DevicesPage.tsx [hunk 5]
```diff
@@ -37,8 +45,8 @@ export const DevicesPage = () => {
           <div className={css.header}>
             <SearchField />
             <AccountSelect />
-            <FilterButton />
             <RefreshButton />
+            <FilterButton handleDrawer={handleDrawer} open={open} />
```
Comment (JamieRuderman) [inline frontend/src/pages/DevicesPage/DevicesPage.tsx:49]: Could also be called onOpenDrawer....
```suggestion
            <FilterButton onOpen={onOpen} open={open} />
```

File: frontend/src/components/FilterDrawerContent/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export { FilterDrawerContent } from './FilterDrawerContent'
```
Comment (JamieRuderman) [inline frontend/src/components/FilterDrawerContent/index.ts:1]: I've started just making components not wrapped in folders. Feel free to do the same if so inclined.   🙂

File: frontend/src/components/FilterDrawerContent/FilterSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+          <ListItemText className={css.name} primary={props.filterName} />
+        </Typography>
+      )
+    }
+    return (
+      <Typography>
+        <ListItemText className={css.name} primary={props.filterName} />
+      </Typography>
+    )
+  }
```
Comment (JamieRuderman) [inline frontend/src/components/FilterDrawerContent/FilterSelector.tsx:85]:
```suggestion
  function TheFilter(props: { value: string; filterName: string }) {
    const selected = props.value === owner || props.value === filter || props.value === sort
      return (
        <Typography color={selected ? 'primary' : undefined}>
          <ListItemText className={css.name} primary={props.filterName} />
        </Typography>
      )
  }
```

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
Looks really good overall. Love the feature. Main thing is just to replace the <Divider /> component with I'm thinking the <Collapse /> component. Then the css should all fall into place.
Thanks!

File: frontend/src/components/DeviceList/DeviceList.tsx [hunk 8]
```diff
@@ -6,70 +6,101 @@ import { DeviceSetupItem } from '../DeviceSetupItem'
...
+                  <Notice>This device is registered to another user</Notice>
+                </ListItem>
+                <Divider />
+              </>
+            ))
           ) : (
             <>
               <ListItem>
                 <Notice>This device is not registered to you.</Notice>
               </ListItem>
```
Comment (JamieRuderman) [inline frontend/src/components/DeviceList/DeviceList.tsx:59]:
```suggestion
```

File: frontend/src/components/DeviceList/DeviceList.tsx [hunk 9]
```diff
@@ -6,70 +6,101 @@ import { DeviceSetupItem } from '../DeviceSetupItem'
...
+                  connections={connections[registeredId]}
+                  thisDevice={true}
+                  setContextMenu={setContextMenu}
+                />
+                <Divider />
+              </>
+            ) : (
+              <>
+                <ListItem>
+                  <Notice>This device is registered to another user</Notice>
```
Comment (JamieRuderman) [inline frontend/src/components/DeviceList/DeviceList.tsx:50]:
```suggestion
                  <Notice>This device is not registered to you.</Notice>
```

File: frontend/src/components/DeviceList/DeviceList.tsx [hunk 10]
```diff
@@ -6,70 +6,101 @@ import { DeviceSetupItem } from '../DeviceSetupItem'
...
-            device.id !== myDevice?.id && (
-              <DeviceListItem
-                key={device.id}
-                device={device}
-                connections={connections[device.id]}
-                setContextMenu={setContextMenu}
-              />
+          )}
+          {devices.length ? (
+            devices.map(
```
Comment (JamieRuderman) [inline frontend/src/components/DeviceList/DeviceList.tsx:65]:
```suggestion
        {devices.map(	
```

File: frontend/src/components/DeviceList/DeviceList.tsx [hunk 11]
```diff
@@ -6,70 +6,101 @@ import { DeviceSetupItem } from '../DeviceSetupItem'
...
-        serviceID={contextMenu.serviceID}
-        setEl={el => setContextMenu({ ...contextMenu, el })}
-      />
-    </>
+          ) : (
+            <Body center>
+              <Typography variant="body1" color="textSecondary">
+                Not devices found
+              </Typography>
+            </Body>
```
Comment (JamieRuderman) [inline frontend/src/components/DeviceList/DeviceList.tsx:81]:
```suggestion
```

Check these edits, I think they are probably pulled from an older version of the file and aren't right.

**Review by JamieRuderman — CHANGES_REQUESTED** [review]
Just looks like a merge issue in the DeviceList.tsx, otherwise all looks awesome!
I marked WIP until we are ready to merge.

**Review by JamieRuderman — APPROVED** [review]
Looks great thanks!

**Pull request merged**